### PR TITLE
Don't trace sensitive headers

### PR DESCRIPTION
**Pull request by rkondratowicz**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilter.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import static uk.gov.hmcts.reform.draftstore.service.AuthService.SERVICE_HEADER;
+
+@Component
+public class RequestTraceFilter extends WebRequestTraceFilter {
+
+    public final static List<String> sensitiveHeaders =
+        Arrays.asList(
+            SECRET_HEADER,
+            SERVICE_HEADER
+        );
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilter.java]: Spring does not trace `Authorization` header by default

File: src/main/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilter.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @Override
+    protected void postProcessRequestHeaders(Map<String, Object> headers) {
+        headers
+            .keySet()
+            .stream()
+            .filter(header -> shouldBeRemoved(header))
+            .collect(toSet())
+            .forEach(h -> headers.remove(h));
```
Comment (KarolJastrzebski) [inline src/main/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilter.java]:
```java
            .filter(this::shouldBeRemoved)
            .collect(toSet())
            .forEach(headers::remove);
```

File: src/main/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilter.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    private boolean shouldBeRemoved(String header) {
+        return sensitiveHeaders
+            .stream()
+            .anyMatch(sensitiveHeader -> header.equalsIgnoreCase(sensitiveHeader));
```
Comment (KarolJastrzebski) [inline src/main/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilter.java]:
```java
            .anyMatch(header::equalsIgnoreCase);
```

File: src/test/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilterTest.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.draftstore.service.AuthService.SECRET_HEADER;
+import static uk.gov.hmcts.reform.draftstore.service.AuthService.SERVICE_HEADER;
+
+public class RequestTraceFilterTest {
+
+    @Test
+    public void should_filter_out_sensitive_headers() throws Exception {
```
Comment (KarolJastrzebski) [inline src/test/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilterTest.java]: unnecessary `throws Exception`

File: src/main/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilter.java (hunk 2 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilter.java]: Personally, I like lambdas more, as they provide slightly more information (by naming the parameter), which makes it easier to figure out what's going on. Probably a matter of preference, though.

File: src/main/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilter.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        super(repository, properties);
+    }
+
+    @Override
+    protected void postProcessRequestHeaders(Map<String, Object> headers) {
+        headers
+            .keySet()
+            .stream()
+            .filter(header -> shouldBeRemoved(header))
+            .collect(toSet())
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilter.java]: Is this line needed?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilter.java]: don't want to remove items while iterating over a collection

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/draftstore/config/RequestTraceFilter.java]: 👍

**Pull request merged**